apis/github

```diff
@@ -1,14 +1,14 @@
 local JSON = dofile("apis/dkjson")
 
--- Build a github API url, trying to authenticate (not tested)
+local function generateAuthHeaders(auth)
```

Comment:
@eric-wieser Sorry your comment got deleted because I rebased, I won't do that anymore until this passes CR.  This was moved into a separate function because it's called in both `getAPI` and `Auth.checkToken()`